### ISSUE TITLE
Regression in template interpolation post 1.7.1 with hash merging enabled

### ISSUE THREAD

**Issue by hfinucane** [ansible/ansible#11053]
Minimum test play:

```
- hosts: all
  roles:
  - role: test
    x: "{{adict}}"
```

with hash merging on, and this in the group vars:

```

---

adict:
    foo: bar
```

Resulting error message:

```
fatal: [localhost] => failed to combine variables, expected dicts but got a 'dict' and a 'unicode'
```

I git bisected this back to 9a0f8f0, a fix for #9497. It works with hash merging off.

**Comment by bcoca** [comment]
weird, this should not merge vars at all, just substitute

**Comment by d-a-n** [comment]
Is there any known workaround/hotfix atm?

**Comment by halberom** [comment]
Came across this trying to help someone on IRC, I traced it through to 
https://github.com/ansible/ansible/blob/devel/lib/ansible/runner/__init__.py#L643
https://github.com/ansible/ansible/blob/devel/lib/ansible/utils/__init__.py#L1492
https://github.com/ansible/ansible/blob/devel/lib/ansible/utils/__init__.py#L823

```
> /lib/ansible/utils/__init__.py(824)merge_hash()->None
-> result[k] = merge_hash(a[k], v)
(Pdb) p result[k]
{'foo': 'bar'}
(Pdb) p k
'x'
(Pdb) p a[k]
{'foo': 'bar'}
(Pdb) p v
u'{{ adict }}'
(Pdb) p result
{'delegate_to': None, 'failed_when': None, ...}
```

**Comment by hfinucane** [comment]
Has anyone tried it on v2?

**Comment by d-a-n** [comment]
@hfinucane yes, same error on v2

**Comment by hfinucane** [comment]
On devel, it works for me in v2, but not if I source from the 'v1' folder.

**Comment by bcoca** [comment]
v1 folder won't be released, it is there for convenience of those that use devel normally and find a showstopper in the current head.

So fixed in current devel?

**Comment by hfinucane** [comment]
Yup. Are there any future 1.9 patches planned? Or is this now in wontfix territory?

**Comment by bcoca** [comment]
not sure, need to find out what fixes it in v2 first, then see if we can easily pass to 1.9.x (it probably won't make 1.9.2)

**Comment by d-a-n** [comment]
https://github.com/ansible/ansible/commit/30c1a2d86192fedc706b43a76c26c6e4c31a6fe0 fixes this issue and #11098

A backport to 1.9 would be great.

**Comment by bcoca** [comment]
sadly the commit is incompatible and only works due to the v2 rewrite. closing this as it is fixed in devel
